frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/rbac.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  AccessReviewResourceAttributes,
+  K8sVerb,
+  SelfSubjectAccessReviewKind,
+} from '../../../extensions/console-types';
+import { ProjectModel, SelfSubjectAccessReviewModel } from '../../../models';
+import { k8sCreate } from '../../../utils/k8s/k8s-resource';
+import { getActiveCluster, getImpersonate } from '../../core/reducers/coreSelectors';
+import { ImpersonateKind } from '../../redux-types';
+import storeHandler from '../../storeHandler';
+import { useSafetyFirst } from '../safety-first';
+
+// Memoize the result so we only make the request once for each access review.
+// This does mean that the user will have to refresh the page to see updates.
+// Accept an `impersonateKey` parameter to include in the cache key even though
+// it's not used in the function body. (Impersonate headers are added
+// automatically by `k8sCreate`.) This function takes in the destructured
+// resource attributes so that the cache keys are stable. (`JSON.stringify` is
+// not guaranteed to give the same result for equivalent objects.)
+const checkAccessInternal = _.memoize(
+  (
+    group: string,
+    resource: string,
+    subresource: string,
+    verb: K8sVerb,
+    name: string,
+    namespace: string,
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    impersonateKey: string,
+  ): Promise<SelfSubjectAccessReviewKind> => {
+    // Projects are a special case. `namespace` must be set to the project name
+    // even though it's a cluster-scoped resource.
+    const reviewNamespace =
+      group === ProjectModel.apiGroup && resource === ProjectModel.plural ? name : namespace;
+    const ssar: SelfSubjectAccessReviewKind = {
+      apiVersion: 'authorization.k8s.io/v1',
+      kind: 'SelfSubjectAccessReview',
+      spec: {
+        resourceAttributes: {
+          group,
+          resource,
+          subresource,
+          verb,
+          name,
+          namespace: reviewNamespace,
+        },
+      },
+    };
+    return k8sCreate(SelfSubjectAccessReviewModel, ssar);
+  },
+  (...args) => [...args, getActiveCluster(storeHandler.getStore().getState())].join('~'),
+);
+
+/**
+ * It provides impersonation key based on data from the redux store.
+ * @param impersonate impersonation details
+ * @return The impersonation key.
+ * * */
+export const getImpersonateKey = (impersonate: ImpersonateKind): string => {
```

Comment:
```suggestion
/**
 * It provides impersonation key based on data from the redux store.
 */
const getImpersonateKey = (impersonate: ImpersonateKind): string => {
```
We don't need to export this -- it's an internal use only. Also we can trim out the additional data here as it's an internal use function.

Reply:
Ahh, good point, thank you